library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -350,6 +350,11 @@ private boolean updateLocked() {
             }
 
             if (!isActive && downloadReadyChecker.canDownload(downloadBatch)) {
+                boolean isBatchStartingForTheFirstTime = batchRepository.isBatchStartingForTheFirstTime(downloadBatch.getBatchId());
```

Comment:
i've put this check here cause it will be triggered only once per batch. so the impact should be minor.